src/_P166_GP8403.ino

```diff
@@ -214,7 +214,7 @@ boolean Plugin_166(uint8_t function, struct EventStruct *event, String& string)
 
     case PLUGIN_INIT:
     {
-      if (Settings.isI2CEnabled()) {
+      if (Settings.isI2CEnabled(get3BitFromUL(Settings.I2C_Flags[event->TaskIndex], I2C_FLAGS_BUS_NUMBER))) {
```

Comment:
Maybe add another version of the `Settings.isI2CEnabled` function with taskindex as argument?
All in this line of code is present in the `Settings` struct.

Or maybe even add the event pointer as argument...

Reply:
Yep, this is going to be a new method `getI2CInterface(taskIndex_t TaskIndex)`, on the `Settings` struct.